src/main/java/com/bakdata/common_kafka_streams/util/TopologyInformation.java

```diff
@@ -42,6 +43,37 @@ public TopologyInformation(final Topology topology, final String streamsId) {
         this.streamsId = streamsId;
     }
 
+    private static List<Node> getNodes(final Topology topology) {
+        return topology.describe().subtopologies()
+                .stream()
+                .flatMap(subtopology -> subtopology.nodes().stream())
+                .collect(Collectors.toList());
+    }
+
+    private static Stream<String> getAllSources(final Collection<Node> nodes) {
+        return nodes.stream()
+                .filter(node -> node instanceof Source)
+                .map(node -> (Source) node)
+                .map(Source::topicSet)
+                .flatMap(Collection::stream);
+    }
+
+    private static Stream<String> getAllSinks(final Collection<Node> nodes) {
+        return nodes.stream()
+                .filter(node -> node instanceof Sink)
+                .map(node -> ((Sink) node))
+                .map(Sink::topic);
+    }
+
+    private static boolean isInternalTopic(final String topic) {
+        return topic.startsWith("KSTREAM-") || topic.startsWith("KTABLE-")
+                || topic.endsWith("-repartition") || topic.endsWith("-changelog");
```

Comment:
assuming all topics with those suffixes being internal doesn't seem right. 
For `"-changelog",` you can check if it is a backing topic (i.e. `getBackingTopics(nodes).contains(topic)`). `"-repartition"` would require an analogous method to `getBackingTopics`.

Reply:
Hm. In what case do we consider a topic ending with -repartition not internal?

Reply:
Depending on the definition of internal, I would argue that topics used in `through` are not internal. But obviously they can end with -repartition.

Reply:
What do you think now?